src/twisted/python/test/test_setup.py

```diff
@@ -366,34 +305,33 @@ class LongDescriptionTests(SynchronousTestCase):
 
     def test_generate(self):
         """
-        L{_longDescriptionArgsFromReadme()} outputs a L{long_description} in
+        L{getSetupArgs()} outputs a L{long_description} in
         reStructuredText format. Local links are transformed into absolute ones
         that point at the Twisted GitHub repository.
         """
-        path = self.mktemp()
-        with open(path, "w") as f:
-            f.write(
-                "\n".join(
-                    [
-                        "Twisted",
-                        "=======",
-                        "",
-                        "Changes: `NEWS <NEWS.rst>`_.",
-                        "Read `the docs <https://twistedmatrix.com/documents/>`_.\n",
-                    ]
-                )
-            )
+        path = pathlib.Path(self.mktemp())
+        path.write_text(
+            textwrap.dedent(
+                """\
+                Twisted
+                =======
+
+                Changes: `NEWS <NEWS.rst>`_.
+                Read `the docs <https://twistedmatrix.com/documents/>`_.
+                """
+            ),
+            encoding="utf8",
+        )
 
         self.assertEqual(
-            {
-                "long_description": """\
-Twisted
-=======
-
-Changes: `NEWS <https://github.com/twisted/twisted/blob/trunk/NEWS.rst>`_.
-Read `the docs <https://twistedmatrix.com/documents/>`_.
-""",
-                "long_description_content_type": "text/x-rst",
-            },
-            _longDescriptionArgsFromReadme(path),
+            textwrap.dedent(
+                """\
+                Twisted
+                =======
+
+                Changes: `NEWS <https://github.com/twisted/twisted/blob/trunk/NEWS.rst>`_.
+                Read `the docs <https://twistedmatrix.com/documents/>`_.
+                """
+            ),
+            getSetupArgs(extensions=[], readme=str(path))["long_description"],
```

Comment:
Since `getSetupArgs()` just passes on the result of `_longDescriptionFromReadme()` unchanged, you could instead call the latter directly. That avoids this test case creating an inflated amount of coverage.

Reply:
this is intentional to avoid coupling getSetupArgs to its specific implementation

Reply:
now that _longDescriptionFromReadme is one expression - I refactored it into getSetupArgs